docs/grammar/SolidityLexer.g4

```diff
@@ -85,6 +85,7 @@ SignedIntegerType:
 Storage: 'storage';
 String: 'string';
 Struct: 'struct';
+Transient: 'transient';  // This is not correctly reflecting the implementation in the parser code, where "transient" is not a keyword and can be used as identifiers. This grammar rejects transient as identifiers in most places.
```

Comment:
In the other cases like this above, we were happy with ``// not a real keyword`` as a comment :-).

Reply:
It's fine either way, I just found the difference in verbosity funny :-).

Reply:
Or well, "This grammar rejects transient as identifiers in most places." is not quite true

Reply:
Ah, right. It's because this was actually not like the other things marked `// not a real keyword`. @matheusaaguiar fixed it now but originally it was rejecting identifiers in most places so that comment was true.